subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/junit/CategoryFilter.java

```diff
@@ -83,15 +80,27 @@ private boolean shouldRun(final Description description, final Description paren
         return result;
     }
 
-    private boolean matches(final Class<?> category, final Set<Class<?>> categories) {
-        for (Class<?> cls : categories) {
-            if (cls.isAssignableFrom(category)) {
+    private boolean matches(final Class<?> category, final Set<String> categories) {
+        ClassLoader classLoader = category.getClassLoader();
+        for (String cls : categories) {
+            if (loadClass(classLoader, cls).isAssignableFrom(category)) {
                 return true;
             }
         }
         return false;
     }
 
+    private Class<?> loadClass(ClassLoader classLoader, String className) {
+        try {
+            if (classLoader == null) {
```

Comment:
This is because some implementation uses `null` to represent bootstrap classloader, i.e. `Object.class.getClassLoader()==null`.

Reply:
Your comment here would probably make a good comment in the code :-)

Reply:
Thanks, good advice.